### ISSUE TITLE
UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape.

### ISSUE THREAD

**Issue by cgd306948366** [matterport/Mask_RCNN#749]
E:\Programs\Python\lib\site-packages\tensorflow\python\ops\gradients_impl.py:100: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
Epoch 1/1

When run the traning，the console shows the warning above. How to solve this problem?

**Comment by JoeLogan1981** [comment]
```
import warnings
warnings.filterwarnings('ignore')
```

**Comment by hemikak** [comment]
Check this out https://github.com/tflearn/tflearn/issues/460#issuecomment-407225809

**Comment by waleedka** [comment]
@cgd306948366 You can safely ignore this warning. It's a preemptive warning from TensorFlow when it cannot be certain of the size of the generated tensor.

**Comment by shen338** [comment]
@waleedka I also run into this. But it cannot be ignored because it will terminate my training. My input image size is 1920x1080 with a V100 GPU, 32G memory. 
And also, loading COCO pretrained model to train is OK, but load my own checkpoint does not work.

**Comment by echoTomei** [comment]
> @waleedka I also run into this. But it cannot be ignored because it will terminate my training. My input image size is 1920x1080 with a V100 GPU, 32G memory.
> And also, loading COCO pretrained model to train is OK, but load my own checkpoint does not work.

Hi, I also meet this problem when training maskrcnn, Have you solved this problem?

**Comment by doublemul** [comment]
> > @waleedka I also run into this. But it cannot be ignored because it will terminate my training. My input image size is 1920x1080 with a V100 GPU, 32G memory.
> > And also, loading COCO pretrained model to train is OK, but load my own checkpoint does not work.
> 
> Hi, I also meet this problem when training maskrcnn, Have you solved this problem?

I meet this problem when training maskrcnn, too. Do you know how to solve it?

**Comment by VietnamAIHub** [comment]
I got the issue with training MaskRCNN especially I use nucleus.py for  iris dataset but it shows the error, then it's stop running anyone could help really appreciate for your help

**Note I already using Ignore warning but it still stop running**

C:\Users\admin\.conda\envs\tensorflownew\lib\site-packages\tensorflow_core\python\framework\indexed_slices.py:425: UserWarning: Converting sparse IndexedSlices to a dense
 Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "

**Comment by thsdd** [comment]
I also got the error like Nhiem for the indexed_slices.py while trying to train the Mask R-CNN network from immersivelimit.com on my own dataset.

previous answer:
"""
the reason maybe was the low memory of chrome. I got maybe 50 tabs open at the moment. For batch size =1 image per GPU it worked for 10 steps of the first epoch, then the kernel died. after changing to  20 images per GPU (i got a very bad GPU, regular one, not especially for ML tasks) it just didn't start and died at the beginning.
I changed to a fresh Firefox window (just 1 tab open) and it seems to work well (atm step 56 of 100, epoch 1 of 5).
I got 64 Gb of RAM, so maybe it is a chrome/browser issue.
"""

edit:
-> didn't run for all epochs and steps! So the browser wasn't the issue. 
solution: I changed the image size to 512x512 and it worked well. Previously I had length and width dividable by 6 times (but length != width) (like required for the aktwelve implementation -> also used 512x512 images). 

I hope it helps.

(Win 10 x64)

**Comment by jyothsnavasu** [comment]
I'm facing the same issue. The training process stops altogether similar to the comments [here](https://github.com/matterport/Mask_RCNN/issues/749#issuecomment-446740019). The training process seems to work when I use weights from ImageNet but if does not work when I use my own checkpoint model file (.h5). 

@waleedka could you please provide more information to debug this? Tagging you here since this issue is closed by you. If you would like me to open a new issue, I can do that as well. 

Thank you very much in advance

**Comment by li-fz** [comment]
> I'm facing the same issue. The training process stops altogether similar to the comments [here](https://github.com/matterport/Mask_RCNN/issues/749#issuecomment-446740019). The training process seems to work when I use weights from ImageNet but if does not work when I use my own checkpoint model file (.h5).
> 
> @waleedka could you please provide more information to debug this? Tagging you here since this issue is closed by you. If you would like me to open a new issue, I can do that as well.
> 
> Thank you very much in advance

I meet the same question,but I can't solve it, hope you can give me some help,thanks!

**Comment by venky012** [comment]
Any one able to fix this issue. I am trying to run it on the google colab. My session is terminating after I get this warning.

**Comment by arwaromdhane** [comment]
any help please!!!!

**Comment by LexiRXW** [comment]
I faced the problem right now and I fixed it. 
It turned out that the img files didn't match the label files. E.g., in one of the above folders, there is a file named 1.jpg, but at the same time, there is no file named 1.yaml

**Comment by venky012** [comment]
> I faced the problem right now and I fixed it. It turned out that the img files didn't match the label files. E.g., in one of the above folders, there is a file named 1.jpg, but at the same time, there is no file named 1.yaml

Same case with mine mask rcnn ground truth annotations were not present in the json file for some image files fixing it worked with me.

**Comment by dongcheng1942** [comment]
I came across with this problem too, I solved it by disable the GPU  (os.environ["CUDA_VISIBLE_DEVICES"]="-1") in visualize.py. 
I don't know why it works but it's useful in my project, i am using my own coco dataset too.

**Comment by cccccrj** [comment]
Check this out  https://github.com/keras-team/keras/issues/4365

**Comment by Jasonseti** [comment]
I ran into this problem when creating MaskRCNN. I tried to sample random values from a tensor using tf.gather().

`negative_loss = tf.gather(negative_loss, random_indices)`

That's when Tensorflow spits out those warnings. I then found out that tf.gather() is a deprecated argument and doesn't properly return gradients. Instead, you should use tf.gather_nd, which is supported.

`negative_loss = tf.gather_nd([negative_loss], tf.stack([tf.zeros_like(indices), indices], axis=-1))`

After that, the warnings are gone and it trains so much faster.
